packages/material-ui-lab/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -214,7 +214,7 @@ export interface UseAutocompleteProps<
   /**
    * Array of options.
    */
-  options: Array<T | Option>;
+  options: T[] | AutocompleteOption[];
```

Comment:
How about having it like this? This will prevent people from mix and matching different data types inside the array.

Reply:
Right, the follow prop-types isn't an option for us:

```tsx
  /**
   * Array of options.
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
    }),
  ).isRequired,
```

I would probably expect

```jsx
  /**
   * Array of options.
   */
  options: PropTypes.any,
```

Reply:
This would allow passing `AutocompleteOption` while `onChange` thinks it's receiving `T`, no?